sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -33,8 +33,8 @@ import org.apache.spark.util.collection.BitSet
  * Performs a sort merge join of two child relations.
  */
 case class SortMergeJoinExec(
-    leftKeys: Seq[Expression],
-    rightKeys: Seq[Expression],
+    var leftKeys: Seq[Expression],
```

Comment:
What information are you missing? The SortMergeExec is replaced after each planning iteration.

I would prefer that we use a `lazy val` here instead.